### PR TITLE
Refactor substructure CLI more like corresponding API endpoint

### PR DESCRIPTION
**Pull request by JJ-Pineda**

### CODE REVIEW

File: src/command_line/substructure_search.rs [hunk 1]
```diff
@@ -32,61 +33,86 @@ pub fn command() -> Command {
...
-    let limit = if let Some(limit) = limit {
-        limit.parse::<usize>()?
+    let result_limit = if let Some(result_limit) = result_limit {
+        result_limit.parse::<usize>()?
     } else {
         usize::try_from(1000)?
     };
 
-    let tantivy_result_limit = limit * 10;
+    let tautomer_limit = &result_limit * 10;
```
Comment (xrl) [inline src/command_line/substructure_search.rs:51]: Try running `cargo clippy` and see if it complains about this line. `result_limit` is a usize which is "trivially" copyable in rust speak so you wouldn't bother with a reference. The reference will be 64bit and the usize is 64bit too so it's just extra indirection.

**Review by xrl — COMMENTED** [review]
Very cool work, I can tell you're enjoying have a compiler telling you what's up. Refactor boldly!

I just have the one bit about the `&` that snuck in there. If you can confirm what clippy says and make sure clippy is happy I can turn around a quick approve.

File: src/command_line/substructure_search.rs (hunk 1 above)
Comment (JJ-Pineda) [reply src/command_line/substructure_search.rs]: Forgot all about clippy... Should be all set now!

**Review by xrl — APPROVED** [review]
Looks good!

**Pull request merged**